modules/build/config.nix

```diff
@@ -12,8 +12,8 @@ with lib;
 
     build = {
       arch = mkOption {
-        type = types.enum [ "aarch64" ];
-        default = "aarch64";
+        type = types.enum [ "aarch64" "x86_64" ];
+        default = strings.removeSuffix "-linux" builtins.currentSystem;
```

Comment:
We should not add impure default values. We can ensure backwards compatibility with this for stateVersion <= 24.05 but should get rid of this impurity in 24.11.

Or we just use nixpkgs.system instead of this property everywhere. Not sure why we are using this arch option instead of just the nixpkgs.system